PWGHF/Tasks/taskMcEfficiencyToXiPi.cxx

```diff
@@ -0,0 +1,435 @@
+
+  ConfigurableAxis axisPt{"axisPt", {200, 0, 20}, "pT axis"};
+  ConfigurableAxis axisMass{"axisMass", {900, 2.1, 3}, "m_inv axis"};
+
+  Service<o2::framework::O2DatabasePDG> pdg;
+  HfHelper hfHelper;
+
+  enum HFStep { kHFStepMC = 0,
+                kHFStepMcInRapidity,
+                kHFStepAcceptance,
+                kHFStepTrackable,
+                kHFStepAcceptanceTrackable,
+                kHFStepItsTpcTrackableCuts,
+                kHFStepItsTrackableCuts,
+                kHFStepTracked,
+                kHFStepTrackedCuts,
+                kHFStepTrackedSelected,
+                kHFNSteps };
+
+  enum TrackableStep { kTrackableAll = 0,
+                       kTrackableITS,
+                       kTrackableTPC,
+                       kTrackableITSTPC,
+                       kNTrackableSteps };
+
+  using TracksWithSelectionMC = soa::Join<aod::Tracks, aod::TracksExtra, aod::McTrackLabels, aod::TrackSelection>;
+  using candidateInfo = soa::Join<aod::HfCandToXiPi, aod::HfToXiPiMCRec, aod::HfSelToXiPi>;
+  using genLevelInfo = soa::Join<aod::McParticles, aod::HfToXiPiMCGen>;
+
+  HistogramRegistry registry{"registry"};
+
+  void init(InitContext&)
+  {
+    auto hCandidates = registry.add<StepTHn>("hCandidates", "Candidate count at different steps", {HistType::kStepTHnF, {axisPt, axisMass, {2, -0.5, 1.5, "collision matched"}, {RecoDecay::OriginType::NonPrompt + 1, RecoDecay::OriginType::None - 0.5, RecoDecay::OriginType::NonPrompt + 0.5}}, kHFNSteps});
+    hCandidates->GetAxis(0)->SetTitle("#it{p}_{T} (GeV/#it{c})");
+    hCandidates->GetAxis(1)->SetTitle("#it{m}_{inv} (GeV/#it{c}^{2})");
+    hCandidates->GetAxis(2)->SetTitle("Collision matched");
+    hCandidates->GetAxis(3)->SetTitle("Charm hadron origin");
+
+    auto hTrackablePtEta = registry.add<StepTHn>("hTrackablePtEta", "Prongs kinematics at different steps", {HistType::kStepTHnF, {{200, 0, 20}, {400, -2, 2}}, kNTrackableSteps});
+    hTrackablePtEta->GetAxis(0)->SetTitle("#it{p}_{T} (GeV/#it{c})");
+    hTrackablePtEta->GetAxis(1)->SetTitle("#eta");
+  }
+
+  template <typename T>
+  inline bool checkTrackGlbTrk(T track)
+  {
+    bool checkGlbTrk = track.isGlobalTrackWoDCA();
+    bool checkNClsTpc = false;
+    bool checkNClsIts = false;
+    if (track.tpcNClsFound() > nClustersTpcMin) {
+      checkNClsTpc = true;
+    }
+    if (track.itsNCls() > nClustersItsMin) {
+      checkNClsIts = true;
+    }
+    if (checkGlbTrk && checkNClsTpc && checkNClsIts) {
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  template <typename T>
+  inline bool checkTrackItsTrk(T track)
+  {
+    bool checkItsTrk = track.isQualityTrackITS();
+    bool checkNClsIts = false;
+    if (track.itsNCls() > nClustersItsMin) {
+      checkNClsIts = true;
+    }
+    if (checkItsTrk && checkNClsIts) {
+      return true;
+    } else {
+      return false;
+    }
+  }
```

Comment:
This is equivalent to
```c++
track.isQualityTrackITS() && track.itsNCls() > nClustersItsMin
```

Reply:
I meant that the entire function is equivalent to that line. You can directly return the value of the condition instead returning `true` or `false` after evaluating the condition. Plus the function is used only once which makes it unnecessary. You can just use that one-line condition directly instead of calling the function.